datumaro/plugins/cifar_format.py

```diff
@@ -0,0 +1,179 @@
+                label_cat.add(label)
+
+        return { AnnotationType.label: label_cat }
+
+    def _load_items(self, path):
+        items = {}
+
+        image_dir = osp.join(osp.dirname(path), CifarPath.IMAGES_DIR)
+        if osp.isdir(image_dir):
+            images = { osp.splitext(osp.relpath(p, image_dir))[0]: p
+                for p in find_images(image_dir, recursive=True) }
+        else:
+            images = {}
+
+        # 'batch_label': 'training batch 1 of 5'
+        # 'data': ndarray
+        # 'filenames': list
+        # 'labels': list
+        with open(path, 'rb') as anno_file:
+            annotation_dict = pickle.load(anno_file)
+
+        labels = annotation_dict.get('labels')
+        filenames = annotation_dict.get('filenames')
+        images_data = annotation_dict.get('data')
+        size = annotation_dict.get('image_sizes')
+
+        for i, (filename, label, image_data) in \
+                enumerate(zip(filenames, labels, images_data)):
+            item_id = osp.splitext(filename)[0]
+            annotations = []
+            if label != None:
+                annotations.append(Label(label))
+
+            image = images.get(item_id)
+            if not image:
+                image = image_data
+                if size and image.any():
+                    image = image.reshape(size[i][0],
+                        size[i][1], 3).astype(np.uint8)
+                elif image.any():
+                    image = image.reshape(CifarPath.IMAGE_SIZE,
+                        CifarPath.IMAGE_SIZE, 3).astype(np.uint8)
+
+            items[item_id] = DatasetItem(id=item_id, subset=self._subset,
+                image=image, annotations=annotations)
+
+        return items
+
+
+class CifarImporter(Importer):
+    @classmethod
+    def find_sources(cls, path):
+        return cls._find_sources_recursive(path, '', 'cifar',
+            file_filter=lambda p: osp.basename(p) != CifarPath.BATCHES_META and \
+                osp.basename(p) != CifarPath.IMAGES_DIR)
+
+
+class CifarConverter(Converter):
+    DEFAULT_IMAGE_EXT = '.png'
+
+    def apply(self):
+        label_categories = self._extractor.categories()[AnnotationType.label]
+
+        label_names = []
+        for label in label_categories:
+            label_names.append(label.name)
+        labels_dict = { 'label_names': label_names }
+        batches_meta_file = osp.join(self._save_dir, CifarPath.BATCHES_META)
+        with open(batches_meta_file, 'wb') as labels_file:
+            pickle.dump(labels_dict, labels_file)
+
+        for subset_name, subset in self._extractor.subsets().items():
+            filenames = []
+            labels = []
+            data = []
+            image_sizes = {}
+            for item in subset:
+                filenames.append(item.id + self._find_image_ext(item))
+
+                if item.has_image and self._save_images:
+                    self._save_image(item, osp.join(self._save_dir, CifarPath.IMAGES_DIR,
```

Comment:
Do we need an option to pack images into a single file?

Reply:
I think that the ability to convert images to single files can be useful here.